src/main/java/com/insightfullogic/honest_profiler/core/profiles/lean/info/NumericInfo.java

```diff
@@ -0,0 +1,189 @@
+package com.insightfullogic.honest_profiler.core.profiles.lean.info;
+
+import static java.math.BigInteger.ZERO;
+
+import java.math.BigInteger;
+
+import com.insightfullogic.honest_profiler.core.aggregation.result.Aggregation;
+import com.insightfullogic.honest_profiler.core.parser.StackFrame;
+import com.insightfullogic.honest_profiler.core.profiles.lean.LeanNode;
+
+/**
+ * NumericInfo collects the four basic amounts tracked by the profiles and {@link Aggregation}s
+ * <ul>
+ * <li>Self Time, which is the estimated amount of time in nanoseconds spent executing a method (or an aggregated set of
+ * methods), excluding the amount of time spent in methods (in)directly called by the method(s)</li>
+ * <li>Total Time, which is the estimated amount of time in nanoseconds spent executing a method (or an aggregated set
+ * of methods), including the amount of time spent in methods (in)directly called by the method(s)</li>
+ * <li>Self Sample Count, which is the number of sample stack traces in which the code from a method (or an aggregated
+ * set of methods) was seen as being executed</li>
+ * <li>Total Sample Count, which is the number of sample stack traces in which the code from a method (or an aggregated
+ * set of methods) or a method called (in)directly by the method(s) was seen as being executed</li>
+ * </ul>
+ *
+ * The class also provides aggregation methods which aggregate a sample or other NumericInfo into this one.
+ */
+public class NumericInfo
+{
+    // Instance Properties
+
+    private BigInteger selfTime;
+    private BigInteger totalTime;
+
+    private int selfCnt;
+    private int totalCnt;
```

Comment:
Given that this is a number of samples, and that INT_MAX is rather large, it would take a program which samples 1000 times a second nearly 50 days of profiling to overflow. I would not worry.